gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPool.cs

```diff
@@ -74,6 +85,36 @@ private void AddConnection(Connection connection)
             }
         }
 
+        private void ConsiderUnavailable()
+        {
+            CloseAndRemoveAllConnections();
+        }
+
+        private void CloseAndRemoveAllConnections()
+        {
+            lock (_connectionsLock)
+            {
+                TeardownAsync().WaitUnwrap();
+                RemoveAllConnections();
+            }
+        }
+
+        private void RemoveAllConnections()
+        {
+            while (!_connections.IsEmpty)
+            {
+                _connections.TryTake(out var connection);
+                connection.Dispose();
+            }
+        }
+
+        private async Task TeardownAsync()
+        {
+            var closeTasks = new List<Task>(_connections.Count);
```

Comment:
The list isn't necessary, you can use the `Task.WhenAll(IEnumerable<Task>)` overload instead: `Task.WhenAll(_connections.Select(c => c.CloseAsync()))`